R/plot.R

```diff
@@ -440,21 +441,25 @@ plot_predictions <- function(data,
     intervals <- data.table::dcast(intervals, ... ~ boundary,
                                    value.var = "prediction")
 
-    plot <- plot +
-      ggdist::geom_lineribbon(
-        data = intervals,
-        aes(
-          ymin = lower, ymax = upper,
-          # We use the fill_ramp aesthetic for this instead of the default fill
-          # because we want to keep fill to be able to use it for other
-          # variables
-          fill_ramp = factor(range, levels = sort(unique(range), decreasing = TRUE))
-        ),
-        lwd = 0.4
-      ) +
-      ggdist::scale_fill_ramp_discrete(
-        name = "range"
-      )
+    # only plot ranges if there are ranges to plot
+    if (!all(range == 0)) {
+      plot <- plot +
+        ggdist::geom_lineribbon(
+          data = intervals,
+          aes(
+            ymin = lower, ymax = upper,
+            # We use the fill_ramp aesthetic for this instead of the default fill
+            # because we want to keep fill to be able to use it for other
+            # variables
+            fill_ramp = factor(range, levels = sort(unique(range), decreasing = TRUE))
+          ),
+          lwd = 0.4
+        ) +
+        ggdist::scale_fill_ramp_discrete(
+          name = "range",
+          range = c(0.15, 0.75)
```

Comment:
This doesn't seem to produce the expected effect (which I assume was what you mention in your 2nd point). Since it adds complexity to the code, I suggest removing it.

Reply:
you mean the `range = c()` part? if we don't have it, then you often can't see the median line, as it as the ribbon and the line have the same opacity.

Reply:
Could you add a comment saying just that please? Otherwise, we will have forgotten next time we revisit and will either remove it (by mistake) or spend time trying to figure out what it does.